1-labeled-tuples/1-labeled-tuples.md

```diff
@@ -0,0 +1,258 @@
+```
+operand =
+ ...
+ | Label(Label.t)
+ | Prj(UHExp.t, Label.t)
+ ```
+<!-- TODO: similar considerations as above -->
+
+Partially labeled product expressions are allowed, and labels and non-labeled positions can be interleaved.
+For example, the labeled product expression `(.x 2, 3, .y True)` is allowed. 
+
+Singleton label product expressions are supported. For example, `.x 2` is supported.
+
+<!-- TODO: can you omit labels by providing values in order: `(1, 2, 3) <= (.x Num, .y Num, .z Num)` -->
+An unlabeled product expression can analyze to a labeled product expression, allowing you to omit labels. For example, `(1, 2, 3) <= (.x Num, .y Num, .z Num)` is valid and each value in `(1,2,3)`. This operation happens positionally, and does not assign labels based on variable name if variables are used. For example, `(y, x, z) <= (.x Num, .y Num, .z Num)` is valid. 
+
+<!-- 
+TODO: "Record punning" in Reason: `{x, y, z} => {x: x, y: y, z: z}` -- is there anything analogous that we can do? Does this interact with positional values? `(y, x, z) <= (.x Num, .y Num, .z Num)` does that operate positionally or via punning?
+TODO: partially labeled values, where some of the arguments are in order: `(1, 2, .z 3) <= (.x Num, .y Num, .z Num)`. what about interleaving vs. requiring all the explicit labels at the end ala Python?
+TODO: what are the type synthesis and type analysis rules for the labeled tuple expressions -->
+### Punning
+Some languages, such as Reason, have record punning. This is where a record's labels can be implied when it is declared using variables. For example, `{x, y, z} => {x: x, y: y, z: z}` is true. Similar punning fuctionality could be implemented in Hazel. For example, if this punning was implemented `(x, y, z) => (x: x, y: y, z: z)` would hold true. However, this creates a question of if every tuple created with variables should be a labeled tuple. Given this additional complexity and added development costs, I believe that labeled tuple punning should be considered out of scope for labeled tuples. 
+
+### Projection Expressions
+
+Syntax: `e.label`
+
+Files to Edit: UHExp.re
+
+Labels can be used to access elements of a pair through projection expressions. To add projection, the operand type must be extended to include projection.
+<!-- TODO: add `e.label` as a new expression form -->
+```
+operand =
+ ...
+ | Prj(UHExp.t, Label.t)
+ ```
+
+`e.label` will be the new expression form. `e.label` expects `e` to synthesize to a labeled tuple type and `label` to match one of the labels within `e`. 
+
+`e.label` will return the value that has the label `label`.
+#### Backspace
+You can press backspace on `e |.label` and get to `e.label`
+
+You can press space on `e|.label` and get to `e .label`
+
+### Type Sythesis and Type Analysis Rules for Labeled Product Expressions
+#### Synthesis
+![Synthesis Rule for Labeled Product](syn_2.png)
+
+![Synthesis Rule for Projection](syn_1.png)
+
+#### Analysis
+![Analysis Rule Single](ana_1.png)

```

Comment:
don't want to allow a type to be treated as a 1-tuple implicitly, so don't need this rule.